crates/crypto/src/key/type/symmetric.rs

```diff
@@ -0,0 +1,152 @@
+use std::ops::Deref;
+
+use rand::{rngs::OsRng, CryptoRng, RngCore};
+use zeroize::ZeroizeOnDrop;
+
+use crate::{
+    key::KeyMetadata, AlgorithmInstance, Error, SecretKey, SignatureVerification, SigningKey,
+    VerifyingKey,
+};
+
+use super::KeyType;
+
+#[derive(Clone, ZeroizeOnDrop)]
+pub struct SymmetricKey(Box<[u8]>);
```

Comment:
Why Box<[u8]> instead of Vec<u8>?

Reply:
`Vec` provides more capabilities like resizing that we don't need here. So to me the question is more "Why Vec when Box<[u8]> is enough"?

Reply:
Because in most applications an owned slice will be a Vec<u8>, so we're adding unnecessary overhead by requiring conversion between Box<[u8]> and Vec<u8>.

Additional capabilities on a Vec don't actually cause any issues right? There's just an extra byte of memory used to store the capacity of the slice.

Conversion from Vec<u8> to Box<[u8]> requires shrinking the Vec to capacity, manually dropping and reading the vec into a new buffer. I'm not an expert but I think that also implies that the original Vec buffer is not zeroized. So overall I think it makes more sense to take ownership of the Vec.

Reply:
RustCrypo always use fixed-length arrays, not `Vec`, so I'm not worried about shrinking stuff. But it's true that many functions are expecting `Vec`s so I'm not going to fight for it, I'll remove the `Box<[u8]>`s.